Support for themes and new design system.

theskumar opened the pull request:
Implementing theme support with https://daisyui.com

## Test Steps

 - [ ] Test lock and feel of everything

theskumar committed:
Collapse color

theskumar committed:
Update fieldset

theskumar committed:
Update field set

theskumar committed:
Update tinymce and choices

theskumar committed:
Update skeleton color

theskumar committed:
Update export button

theskumar committed:
Update detail page, filter label, text-base-content/80

theskumar committed:
Checkbox django-table

theskumar committed:
Fix button--remove

theskumar committed:
formatting

theskumar committed:
Revisions

theskumar committed:
batch action, action-bar, invoices table

theskumar committed:
update

theskumar committed:
paf status sidebar

wes-otf committed:
Added relative timezones to invoice statuses & timezone info to stamped invoice

theskumar committed:
cleanup list-reveal, modal and invoice-block

theskumar committed:
Improve co-applicant

theskumar committed:
More buttons

theskumar committed:
Fix for tinymce word count

theskumar committed:
Alternate implementation of max-length warning

This will alert for all input with a maxlength set.

Also, update this to show warning only when 80 of length limit is
reached

Currently it is failing on Reminder > Add form.

FYI @frjo

https://github.com/HyphaApp/hypha/commit/689582dd3a1608fee60a3ed4ff407b2446416b01

theskumar committed:
co-applicant & text-gray-500

theskumar committed:
Review button

theskumar committed:
update input to include "time" & "url" type

theskumar committed:
Update hypha-grid--two

theskumar committed:
Account page

theskumar committed:
admin bar and tags section

theskumar committed:
update field css and upload button

theskumar committed:
Fix fields esp filefield

theskumar committed:
todo draft badge

theskumar committed:
More badges

theskumar committed:
border

theskumar committed:
fields, errors and cleanup

theskumar committed:
Fix tab text color

theskumar committed:
sealed submission detail

theskumar committed:
fix

theskumar committed:
Adjustments

theskumar committed:
X - action-bar scss

theskumar committed:
kill status-bar scss

theskumar committed:
Screening buttons

theskumar committed:
error display on requested amount input

theskumar committed:
Upgrade multi-input-field to use alpinejs

theskumar committed:
Update splitdatetime widget

theskumar committed:
textarea validator

theskumar committed:
Update layout and copy - coapplicant accept

theskumar committed:
comment section

theskumar committed:
Report frequecy info

theskumar committed:
determination detail

theskumar committed:
Show fund description on the form

theskumar committed:
Use theme provided primary color

theskumar committed:
applicant dashboard should use admin-bar component

theskumar committed:
New theme for admin bar

theskumar committed:
New theme for footer

theskumar committed:
Updates to co-applicant edit form

theskumar committed:
Adjustment to form

theskumar committed:
Admin bar + container-constrained

theskumar committed:
update

theskumar committed:
Fix address field js

theskumar committed:
updates

theskumar committed:
btn-disabled

theskumar committed:
update

theskumar committed:
Update dashboard round and labs section

theskumar committed:
?? extra markup in table.html

theskumar committed:
dashbord paf for review

theskumar committed:
Update show application on review and determination

theskumar committed:
button--link

theskumar committed:
more card-title

theskumar committed:
project-form page

theskumar committed:
error pages

theskumar committed:
--span-two

theskumar committed:
hijack-bar

theskumar committed:
wrapper--cta-box

theskumar committed:
More wrappers

theskumar committed:
Review sidebar updates

theskumar committed:
translate

theskumar committed:
cleanup buttons

theskumar committed:
submissions

theskumar committed:
Nav colors

theskumar committed:
cleanup wrapper and grid

theskumar committed:
submission sealed hero

theskumar committed:
Remove heading--no-margin

theskumar committed:
header.scss

theskumar committed:
cleanup

theskumar committed:
Test passes now

theskumar committed:
Remove status-block.scss

theskumar committed:
cleanup form--error-inline

theskumar committed:
cleanup defailt h6

theskumar committed:
Remove _table.scss

theskumar committed:
Review List Page update

theskumar committed:
partner dashboard

theskumar committed:
Account and 2FA setup page

frjo commented on hypha/static_src/javascript/esm/choices.js-10-2-0.js:
```diff
@@ -4,5 +4,5 @@
-var e,t={exports:{}};
-/*! choices.js v10.2.0 | © 2022 Josh Johnson | https://github.com/jshjohnson/Choices#readme */window,e=function(){return function(){var e={282:function(e,t,i){Object.defineProperty(t,"__esModule",{value:!0}),t.clearChoices=t.activateChoices=t.filterChoices=t.addChoice=void 0;var n=i(883);t.addChoice=function(e){var t=e.value,i=e.label,r=e.id,s=e.groupId,o=e.disabled,a=e.elementId,c=e.customProper ...
+var e-concontainer-constrained,ts:{}};
+/*! choices.js v10.2.0 | © 2022 Josh Johnson | https://github.com/jshjohnson/Choices#readme */window,e=function(){return function(){var e={282:function(e,t,i){Object.defineProperty(t,"__esModule",{value:!0}),t.clearChoices=t.activateChoices=t.filterChoices=t.addChoice=void 0;var n=i(883);t.addChoice=function(e){var t=e.value,i=e.label,r=e.id,s=e.groupId,o=e.disabled,a=e.elementId,c=e.customProper ...
```
@theskumar  This file should be reverted I think, looks like a mistake in a search and replace.

Get an error from babel when trying to build.

theskumar replied:
I got the error as well, and could not figureout what was wrong, thank you! I'll revert this.

frjo commented on Makefile:
```diff
@@ -25,7 +25,7 @@ help: ## Show this help menu with a list of available commands and their descrip
 serve: .cache/tandem .cache/py-packages .cache/dev-build-fe  ## Run Django server, docs preview, and watch frontend changes
 	@.cache/tandem \
-		'${UV_RUN}python manage.py runserver_plus --settings=$(DJANGO_SETTINGS_MODULE)' \
+		'${UV_RUN}python manage.py runserver 7001 --settings=$(DJANGO_SETTINGS_MODULE)' \
```
Is not runserver_plus a bit better for development?

theskumar replied:
This is accidental change, didn't mean it to land in the PR, I'll remove this. 

That said, I use the `runserver` as the restart experience is much better when multiple files change on the disk, esp. when switching branches or doing multi-file edits. the plus reloads multiple times (one reload for each file changed), in-built server does it only once.

frjo commented:
1. Inconsistent hover state here: ✅ [`6399da2` (#4543)](https://github.com/HyphaApp/hypha/pull/4543/commits/6399da26b0a114755bb3f5df5579ded5a5d965ab)

https://github.com/user-attachments/assets/68ac83b6-741d-41ee-bff7-347ddea266a4

2. Time to rename this to "Comments"? ✅  [`fdff17c` (#4543)](https://github.com/HyphaApp/hypha/pull/4543/commits/fdff17ce64294414ef6fe92881af342f432d4096)

![Skärmavbild 2025-06-03 kl  15 02 20](https://github.com/user-attachments/assets/7726e4fe-5a4f-47ff-835f-0eb6cf2421d1)

3. Missing connecting lines as discussed earlier.

![Skärmavbild 2025-06-03 kl  15 02 30](https://github.com/user-attachments/assets/9f7e418b-0e88-4528-be34-4fb87a2d9ce6)

4. No rounded corners.

![Skärmavbild 2025-06-03 kl  15 03 12](https://github.com/user-attachments/assets/101c51cf-20f4-4c0b-95b9-60bec8f28cff)

5. Borders a bit too subtle I think, not easy to parse the page for a user.

![Skärmavbild 2025-06-03 kl  15 03 39](https://github.com/user-attachments/assets/f84d778c-9186-4fa9-b483-86a1017bf6ec)

6. Prefer the current styly of red indication delete. ✅ 

![Skärmavbild 2025-06-03 kl  14 45 20](https://github.com/user-attachments/assets/ef12c85c-72ae-42f5-bee7-4869f5b3e9ed)

7. In the current design these two have the same styling, I think that is better. Having the "X" on the same side for all would be better as well. ✅ 

![Skärmavbild 2025-06-03 kl  14 47 44](https://github.com/user-attachments/assets/b6edc718-1985-4d82-ba85-5697e027377b)

8. Project list search deserves to be a bit bigger, like in the current design. ✅  [`da3e87f` (#4543)](https://github.com/HyphaApp/hypha/pull/4543/commits/da3e87fc4e33f727a97aba4569b2db8fe46b5bd5)

![Skärmavbild 2025-06-03 kl  14 48 28](https://github.com/user-attachments/assets/bfb7c617-5db9-499d-ace9-ad3796c4d9ac)

9. Pager controls for submission and project lists should look the same. Do not think they need to look like buttons, plainer links would be ok.

![Skärmavbild 2025-06-03 kl  14 48 47](https://github.com/user-attachments/assets/8b0702a2-45b7-4365-ba76-2f5eeda6a480)

![Skärmavbild 2025-06-03 kl  14 48 57](https://github.com/user-attachments/assets/18f1db67-9c6b-445d-8691-cfafa25da577)

10. A lighter background would be better here I think.

![Skärmavbild 2025-06-03 kl  14 49 44](https://github.com/user-attachments/assets/65cd12a9-2b7c-49fb-9c2c-3cf430583815)

11. The current design has underlines here. For me that looks better but we can let staff decide here since they are the ones reading all the submissions. ✅ 

![Skärmavbild 2025-06-03 kl  14 39 55](https://github.com/user-attachments/assets/2f4fd972-b905-47e2-966b-90ae80c151bd)

12. The current state is coloured red in the current design. I think I prefer that but this is a minor point.

![Skärmavbild 2025-06-03 kl  14 40 06](https://github.com/user-attachments/assets/759e564e-f0ed-4fc0-b508-9085888bca2f)

13. The front page list only open calls and even OTF have only five at the moment, most others have less. Therefor I think they can be made bigger and more prominent like in the current design.

![Skärmavbild 2025-06-03 kl  14 40 43](https://github.com/user-attachments/assets/ccfca08e-556e-4838-ad64-f26960b97776)

14. This green is only used here, we can reuse one of the blue colours instead (Variables really and not colours). ✅  [`01fbdc7` (#4543)](https://github.com/HyphaApp/hypha/pull/4543/commits/01fbdc7048c764fc1b2531c9dddfcbfc138a16db)

![Skärmavbild 2025-06-03 kl  14 44 33](https://github.com/user-attachments/assets/68c2a1c5-d1fd-4320-b4d4-e03a45292292)

15. The purple background is too different, feels like to many different colours.  ✅ 

<img width="380" alt="Skärmavbild 2025-06-03 kl  14 49 52" src="https://github.com/user-attachments/assets/289ca738-8fe6-46d9-8ab6-5fc564889f37" />

16. In the current design the background change colour when activating toggle. That makes it easier for users to determine the state.

![Skärmavbild 2025-06-03 kl  14 44 59](https://github.com/user-attachments/assets/554f250b-6ee7-4c01-aa00-2ee111bf12c0)

theskumar commented:
Thank you for the comments, all very good feedback. 

12/ This is minor thing but current implementation doesn't allow for updating the `circle` color in a simple way. The whole line and the circle change color if I pick a different type for a particular stage. I would suggest leave it like this for now. 

5/ The borders subtle b/c of the theme selected the default `light` theme, the `corperate` theme has more high lighted border, i noticed this well, will update it in the custom hypha theme. 

I'll look into other points and make changes as suggested.

theskumar committed:
Updates to reporting-frequency

theskumar committed:
Revert choices.js

theskumar committed:
Feedback1: update lang switcher

theskumar committed:
2/ update "Coversation" tab to "Comments"

theskumar committed:
14/ Fix the color success displayed on the checkbox and radio

theskumar committed:
8/ bigger search input on project and other tables with search

theskumar committed:
15 / deemphasis primary action card

theskumar committed:
11 / rendered answer highlights

theskumar committed:
6 / delete button treatment

theskumar committed:
submission more actions icons

theskumar committed:
6 / clear filter btn update

theskumar committed:
4 / rounded submission tables

theskumar commented:
## Review status

1. ✅  Inconsistent hover state here: [`6399da2` (#4543)](https://github.com/HyphaApp/hypha/pull/4543/commits/6399da26b0a114755bb3f5df5579ded5a5d965ab)

2. ✅ Time to rename this to "Comments"?  [`fdff17c` (#4543)](https://github.com/HyphaApp/hypha/pull/4543/commits/fdff17ce64294414ef6fe92881af342f432d4096)

3. Missing connecting lines as discussed earlier.

4. ✅ No rounded corners. 

5. Borders a bit too subtle I think, not easy to parse the page for a user.

6. ✅  Prefer the current styly of red indication delete. 

7.  ✅ In the current design these two have the same styling, I think that is better. Having the "X" on the same side for all would be better as well. 

8. ✅ Project list search deserves to be a bit bigger, like in the current design.   [`da3e87f` (#4543)](https://github.com/HyphaApp/hypha/pull/4543/commits/da3e87fc4e33f727a97aba4569b2db8fe46b5bd5)
9. Pager controls for submission and project lists should look the same. Do not think they need to look like buttons, plainer links would be ok.

10. ✅  A lighter background would be better here I think.

11. ✅  The current design has underlines here. For me that looks better but we can let staff decide here since they are the ones reading all the submissions. 


12. ✅ The current state is coloured red in the current design. I think I prefer that but this is a minor point.


13. ✅ The front page list only open calls and even OTF have only five at the moment, most others have less. Therefor I think they can be made bigger and more prominent like in the current design.


14. ✅  This green is only used here, we can reuse one of the blue colours instead (Variables really and not colours). [`01fbdc7` (#4543)]
15. ✅  The purple background is too different, feels like to many different colours.  
16. In the current design the background change colour when activating toggle. That makes it easier for users to determine the state.

theskumar committed:
update the pagination button style

theskumar committed:
Remove _nav.scss, update is_active logic

theskumar committed:
Delete button--secondary, button--white

theskumar committed:
Remove button--filters

theskumar committed:
Reimplement cookie consent using alpine and localstorage

theskumar committed:
Minor update

theskumar committed:
Merge remote-tracking branch 'origin/main' into themes

* origin/main:
  Expand co-applicant feature to projects (#4552)
  Set label in project header to "Project lead:" instead of only "Lead:" (#4570)
  Update Django to 4.2.2 and also update other python packages. (#4571)

theskumar committed:
update requirements

theskumar committed:
Tweaks to co-applicants/project permissions UI

theskumar committed:
Update table pagination

theskumar committed:
cleanup

theskumar committed:
Remove _heading.scss

theskumar committed:
Remove _button.scss

theskumar committed:
table-checkbox mixin

theskumar committed:
new adminbar using django-cotton

theskumar committed:
Revert makefile change

theskumar committed:
update separator-dot

theskumar committed:
adminbar update existing stuff

theskumar committed:
Use @media for js detection

theskumar committed:
Fix alpine stuff

theskumar committed:
cleanup

theskumar committed:
Adjust hero header

theskumar committed:
update, better link, fixes

theskumar committed:
Merge branch 'main' into themes

* main:
  Update Django to 4.2.3 and also update other python packages. (#4582)
  Handle rare event of file uploads having data without id (#4581)
  Create a json output version of the open calls on the front page. (#4554)
  Correct project template urls. (#4575)
  Fix submission filters queryset issue (#4572)

theskumar committed:
typography

theskumar committed:
cleanup base.scss

theskumar committed:
Typography round 2

theskumar committed:
update invalid link page

theskumar committed:
Remove wrapper--small and also update the two-factor pages

theskumar committed:
Merge branch 'main' into themes

* main:
  Update to python 3.12.11 (#4584)

theskumar committed:
Remove review-sidebar__split

theskumar committed:
clean up submission table child row feature

theskumar committed:
misc updates

theskumar committed:
.section-header and rounded tables

theskumar committed:
Fixup contracting dashboard

theskumar committed:
fix filter-bar overflow

theskumar committed:
table link formatting

theskumar committed:
updates

theskumar committed:
Footer with theme selector

theskumar committed:
Relative time for subitted in tables

theskumar committed:
cleanup scss

theskumar committed:
cleanup unused status block views, wrapper--bottom and update open/close rounds

theskumar committed:
staff dashboard table border

theskumar committed:
doc-block, contract and project detail sections

frjo commented:
@theskumar I have merged in everything that was on test now so you have some minor merge conflicts.

Rebase of main and we can get this on test.

sandeepsajan0 committed:
Add Author to report

theskumar committed:
Merge remote-tracking branch 'origin/main' into themes

* origin/main:
  Fix report file path issue and duplicate file issue, Fix file path issue for Project and SOW Form (#4580)
  Use xhtml2pdf inplace of reportlab for submission pdf export (#4577)
  Fix co-applicant permission for Partners, Reviewers, and Community Reviewers (#4583)
  Migrate django-fsm to django-viewflow (#4561)
  Delete application comments with the application (#4585)

theskumar committed:
Icon enhancements

theskumar committed:
More updates to project detail page

theskumar committed:
more scss variable cleanup

theskumar committed:
show/hide password feature

theskumar committed:
cleanup

theskumar committed:
password field - title and translations

theskumar committed:
Merge remote-tracking branch 'origin/main' into themes

* origin/main:
  Fix ScoreAnswerWidget(Textarea) browser validation issue with tinymce (#4588)

theskumar committed:
fix comment timeline

theskumar committed:
Remove wrapper__medium

sandeepsajan0 commented:
@theskumar I noticed a small accessibility issue with the login fields. On pressing enter from any of the fields username or password, it is toggling the show password instead of moving forward or Login button.
If I press enter after entering password, it should act as click login button?

https://github.com/user-attachments/assets/1bbbec12-d062-4922-88eb-b9eef4fa361d

sandeepsajan0 committed:
Show report author and date similar to submissions

sandeepsajan0 committed:
Fix updated language

theskumar committed:
Reminder and reminder delete page updates

theskumar committed:
Update report form

theskumar committed:
Update review sidebar styling

theskumar committed:
Remove wrapper--outer-space-medium

theskumar committed:
update to application detail page

theskumar committed:
no-print and manage cookies in footer

theskumar committed:
Fix password show/hide

theskumar commented:
> @theskumar I noticed a small accessibility issue with the login fields. On pressing enter from any of the fields username or password, it is toggling the show password instead of moving forward or Login button. If I press enter after entering password, it should act as click login button?
> 
>  Screen.Recording.2025-06-22.at.4.04.06.PM.mov

@sandeepsajan0 should be fixed now, please check.

frjo commented:
This is now on test site now. 

There are some failing test, "test_active_invoices_with_invoices_in_correct_state" etc., that would be good to get sorted.

wes-otf committed:
Better handling for users entering long names (#4590)

<!--
Thanks for contributing to Hypha!

Please ensure your contributions pass all necessary linting/testing and
that the appropriate documentation has been updated.
-->

<!--
Describe briefly what your pull request changes. If this is resolving an
issue, please specify below via "Fixes #<Github Issue ID>"
-->
Fixes #4589. This adds a global max character limit to the `full_name`
field, working both in the application form and the account settings.
There's a few other fixes also bundled in this:
- Better handling for displaying long names in application details,
submission all (table & list views), etc
- In account view, the slack max input field not showing character count
 - Added translation to a few strings


## Test Steps
<!-- 
If step does not require manual testing, skip/remove this section.

Give a brief overview of the steps required for a user/dev to test this
contribution. Important things to include:
 - Required user roles for where necessary (ie. "As a Staff Admin...")
- Clear & validatable expected results (ie. "Confirm the submit button
is now not clickable")
- Language that can be understood by non-technical testers if being
tested by users
-->

 - [ ] Set `FORCE_LOGIN_FOR_APPLICATION = False`
- [ ] Ensure when submitting an application and not logged in, you
cannot submit a name longer than 255 chars
- [ ] Ensure character count updates on typing for both name & slack
fields in the account view

wes-otf commented:
Something I noticed today: my system default is dark theming but when loading hypha it only flashes dark then goes to light. This is on firefox

(included github to show it was detecting the sys preference)

https://github.com/user-attachments/assets/7dae7d78-cc93-4815-88b1-89705e4959ef

I also can't set it manually in firefox's dev tools

https://github.com/user-attachments/assets/0de13dfc-76bb-410d-bbac-dbbaf463fc0f

theskumar commented:
> Something I noticed today: my system default is dark theming but when loading hypha it only flashes dark then goes to light. This is on firefox
> 
> (included github to show it was detecting the sys preference)
> 
>  Screen.Recording.2025-06-23.at.11.29.14.mov 
> I also can't set it manually in firefox's dev tools
> 
>  Screen.Recording.2025-06-23.at.11.33.46.mov

The theme switcher and default system preference is something I want to work on next, i had hacked a quick switcher, which is not ideal.

wes-otf commented:
ah okay sweet, I didn't realize that wasn't included yet! the current switcher works fine then

frjo committed:
Minor adjustments to report template.

frjo committed:
Merge branch 'feature/gh-4565-report-author' into themes

frjo commented:
This is really taking shape!

1. When Google login is present the buttons do not align.

<img width="648" alt="Skärmavbild 2025-06-23 kl  12 07 03" src="https://github.com/user-attachments/assets/660b4ccd-e4d2-4448-8296-fe801db2f183" />

2. On submissions all view we can switch between list and table view. Displaying these links as tabs feels a bit off. Perhaps show both of them so they really feel like tabs? Or display them more link links/buttons?

![Skärmavbild 2025-06-23 kl  12 08 50](https://github.com/user-attachments/assets/37814a6e-7895-43b7-bc39-41e0412d7bd4)

![Skärmavbild 2025-06-23 kl  12 08 59](https://github.com/user-attachments/assets/1b91c73b-ee7d-4434-808f-276f75d89154)

3. Page margins jumps around a bit at different sizes. I would prefer a more smooth behaviour up until the max width. At some screen sizes you now have wasted space. Like in the first image below.

<img width="634" alt="Skärmavbild 2025-06-23 kl  12 10 25" src="https://github.com/user-attachments/assets/d2611ccf-7bf9-440e-a05c-504d9a54c202" />

<img width="626" alt="Skärmavbild 2025-06-23 kl  12 10 36" src="https://github.com/user-attachments/assets/3bc3f176-2022-4f65-8745-a2c54a17fbcf" />

theskumar committed:
Update SSO login button alignment

theskumar committed:
submission all list and table view switcher

theskumar committed:
update container-constrained behaviour

theskumar commented:
> This is really taking shape!
> 
> 1. When Google login is present the buttons do not align.
> 2. On submissions all view we can switch between list and table view. Displaying these links as tabs feels a bit off. Perhaps show both of them so they really feel like tabs? Or display them more link links/buttons?
> 
> 3. Page margins jumps around a bit at different sizes. I would prefer a more smooth behaviour up until the max width. At some screen sizes you now have wasted space. Like in the first image below.

These three should be resolved now.

frjo commented:
> These three should be resolved now.

Looks good!

But something has happened to the submission and project detail views. They have become very narrow and the sidebar is on top of the main content. Some minor css issue.

wes-otf commented:
A few more things I noticed:

In dark mode, the deadline is not visible
![Screenshot 2025-06-24 at 09 38 08](https://github.com/user-attachments/assets/a9da7721-14c4-47ae-a36f-5519b9e9faa7)

Button text in application detail seems to be different sizes (not sure if this is intentional or not)
![Screenshot 2025-06-24 at 11 34 55](https://github.com/user-attachments/assets/d8e88dc9-e8b1-4e03-bf72-fa73b2bf153a)

Parenthesis around the `Updated` portion
![Screenshot 2025-06-24 at 11 35 13](https://github.com/user-attachments/assets/e6660d66-3d65-4da4-baec-11ab2cd69d79)

theskumar committed:
Fixes - home deadline, btn-lg and updated at

theskumar commented:
> A few more things I noticed:
> 
> In dark mode, the deadline is not visible 
> 
> Button text in application detail seems to be different sizes (not sure if this is intentional or not)
> 
> Parenthesis around the `Updated` portion 

Fixed: [`d091328` (#4543)](https://github.com/HyphaApp/hypha/pull/4543/commits/d0913284146a6d1ea58c72ca2696be393eea5c75)

theskumar committed:
update typography on the forms

theskumar commented:
@frjo possible we can do an auto-deploy from this branch to test environment for sometime?

PS: Please deploy latest, thank you! :)

theskumar committed:
Update to cookiebanner

theskumar committed:
Update application detail - line-clamp

frjo commented:
> @frjo possible we can do an auto-deploy from this branch to test environment for sometime?

I should have thought about that, fixed now.

theskumar committed:
Update submission-text-cleanup

make it generic that applies to all the proses instead being limited to
submission-text.

Also, make it globally available, so this applies to all the prose
everywhere

theskumar committed:
Move more things to tinymce.css

theskumar commented:
@frjo can you take a look at this change https://github.com/HyphaApp/hypha/pull/4543/commits/80a2fd6f81759c5b60ad7ae76ee9bb7b16d6f261 please.

theskumar committed:
Update determination details

theskumar committed:
More update detail and submission answer preview

frjo commented on hypha/templates/base.html:
```diff
@@ -181,29 +180,28 @@
         <!-- Needed for most forms so load it here once. -->
         <script src="{% static 'file_form/file_form.min.js' %}"></script>
         <script src="{% static 'js/file-uploads.js' %}"></script>
+        <script defer src="{% static 'js/behaviours/prose-cleanup.js' %}"></script>
```
@theskumar This can be moved up to "head". Looks neater I think.

theskumar replied:
sure.

frjo commented:
> @frjo can you take a look at this change [80a2fd6](https://github.com/HyphaApp/hypha/commit/80a2fd6f81759c5b60ad7ae76ee9bb7b16d6f261) please.

The `prose-cleanup.js` looks good. Should do the same things as the old `submission-text-cleanup.js` but more solid it looks.

This was only ever used for rich text in submissions. Mainly to slightly improve the spagetti code you get when pasting in from MS Word and similar. Likely no issue running it on all "prose".

theskumar committed:
update review detail and list page

wes-otf commented:
A few other small things:

When pressing submit on an application, all buttons at the bottom turn into `Submitting...` not sure if that's intended or if it should only be the `Submit` button

https://github.com/user-attachments/assets/975837ef-67b8-44cd-ac1a-7b5d1ff15605

In the revision compare view the headings for prompts outside of `Email`, `Name`, and `Requested Funding` don't appear as headings but inline text (`Project description`, `Objectives`):
![Screenshot 2025-06-25 at 10 17 45](https://github.com/user-attachments/assets/afb0a448-4137-48e8-b0bd-bf6107dbfb5f)

theskumar committed:
More relative datetime in tables, using RelativeTimeColumn

theskumar commented:
> A few other small things:
> 
> When pressing submit on an application, all buttons at the bottom turn into `Submitting...` not sure if that's intended or if it should only be the `Submit` button

@frjo do you know more on this? I believe this is an old feature/bug. Happy to change the loading indicator to be more specific.

theskumar commented:
> A few other small things:
> [..]
> In the revision compare view the headings for prompts outside of `Email`, `Name`, and `Requested Funding` don't appear as headings but inline text (`Project description`, `Objectives`): 

Interesting will have dig deeper, do you know how that look in older version or production?

theskumar committed:
Move prose-cleanup.js up in header

theskumar commented:
@wes-otf @frjo I've re-done the review detail page, let me know if there is anything you don't like about the new version.

theskumar committed:
fix application edit typography

theskumar committed:
refactor markdown editor css

wes-otf commented:
> do you know how that look in older version or production?

@theskumar this was from OTF prod:

<img width="1269" alt="Screenshot 2025-06-25 at 13 37 29" src="https://github.com/user-attachments/assets/9d61e2a6-5331-4d72-919f-56622a9bf911" />

theskumar committed:
fix dark mode on field field rendering

theskumar committed:
Remove more .wrapper

frjo commented:
> When pressing submit on an application, all buttons at the bottom turn into Submitting... not sure if that's intended or if it should only be the Submit button
> @frjo do you know more on this? I believe this is an old feature/bug. Happy to change the loading indicator to be more specific

I have seen this but missed to create an issue for it. I think it looks better when only the button clicked say "Submitting…".

frjo commented:
> @wes-otf @frjo I've re-done the review detail page, let me know if there is anything you don't like about the new version.

Overall look good but for the opinion section at the bottom.

![Skärmavbild 2025-06-26 kl  09 52 50](https://github.com/user-attachments/assets/e2e7c906-cb54-4344-8aec-8ec74d1a4de3)

The icon is a bit small, it's too close to the button and the text would look better with text-wrap balanced.

frjo commented:
@theskumar Noticed a bit of left/right instead of start/end has creeped back in.

Classes like pr-1, ml-2 etc should be replaced with pe-1 and ms-2.

frjo commented:
@theskumar Just merged in some Python package updates so need a rebase again.

wes-otf commented:
The longer I use this the more appreciation I have for it - definitely makes Hypha more uniform & slick. Extremely nice work overall @theskumar!

Was fixing a bug with async exports and noticed the download indicator was a little buggy - seems like the green indicator only shows for a second on click and doesn't have much attention drawn to it otherwise to show the user that generation is done. Happy to take a pass at this in my async export fix if you'd prefer!

https://github.com/user-attachments/assets/72d36fa0-99de-465c-832b-f2dd65cc8a7c

theskumar committed:
Fix compare view

theskumar committed:
Remove rich-text--answers

theskumar committed:
update user-menu colors

theskumar committed:
fix container-constrained

theskumar committed:
new layout.css component

theskumar committed:
Invoice delete confirmation modal

theskumar committed:
New submission revision list view

theskumar committed:
Merge remote-tracking branch 'origin/main' into themes

* origin/main:
  Rename "Meta terms" to "Tags" in message on submission edit (#4593)
  Updated python packages June 2025. (#4591)

theskumar commented:
@wes-otf @frjo Implemented new html based diff for revision changes. [`ed4c419` (#4543)](https://github.com/HyphaApp/hypha/pull/4543/commits/ed4c4193497b3645d262aeb4406d30185fe94874)

Let me know what you think? Does this work or you would like old version?

theskumar committed:
tweaks to revision detail view

sandeepsajan0 committed:
Restrict permission to invite and update co-applicant in archived submissions and complete projects

theskumar committed:
update invoice-form page

wes-otf commented on hypha/apply/funds/templates/funds/includes/admin_primary_actions.html:
```diff
@@ -15,123 +15,141 @@ <h5>{% trans "Actions to take" %}</h5>
+        {% if PROJECTS_ENABLED %}
+            {% if object.accepted_for_funding and not object.project %}
+                <button class="btn btn-primary btn-block"  hx-get="{% url 'funds:submissions:create_project' pk=object.pk %}" hx-target="#htmx-modal">
+                    {% trans "Create Project" %}
```
```suggestion
                    {% trans "Create project" %}
```

Super small but the rest of the buttons only capitalize the first word

theskumar replied:
Thanks, resolved

theskumar committed:
update review detail

theskumar committed:
fix for update reviewer modal form

frjo commented:
@theskumar The translations works as they should but we have one UI bug.

The `#translate-card-loading` box is displayed/try to load even when there is no translation happening.

It should only show after the user has opened and submitted the translations form and the url has a lang query like this `/apply/submissions/1761/?fl=en&tl=fr`. 

<img width="1118" alt="Skärmavbild 2025-06-30 kl  17 00 44" src="https://github.com/user-attachments/assets/f8b41320-7c78-4318-94dc-fbb3a8f5c9d3" />

theskumar committed:
update html-diff.css

theskumar committed:
update modal title to use cotton

theskumar committed:
update review opinion delete confirm to dialog

theskumar committed:
Add steps-sm and apply to status bars

theskumar committed:
Update daisyui version

theskumar committed:
tweak html-diff

theskumar committed:
fixup

bickelj commented on hypha/apply/activity/templates/activity/ui/activity-comment-item.html:
```diff
@@ -1,21 +1,21 @@
-            <div class="inline-flex relative justify-center items-start me-4">
-                <div class="relative rounded-full inline-flex items-center justify-center border-white border-2 -ms-0.5 {% if activity.user.is_staff %}bg-slate-200 {% else %}bg-gray-100{% endif %}">
+            <div class="inline-flex relative z-0 justify-center items-start me-4">
+                <div class="relative rounded-full inline-flex items-center justify-center border-white border-2 -ms-0.5 {% if activity.user.is_staff %}bg-slate-200 {% else %}bg-base-200{% endif %}">
```
`border-white` looks like an obsolete color-named class. Is `bg-slate-200` also (is "slate" a color)?

wes-otf replied:
I believe both are tailwind-isms: [`border-white`](https://tailwindcss.com/docs/border-color) and [slate](https://tailwindcss.com/docs/colors)

bickelj replied:
My assumption is this. In order for themes to fully take effect via a few CSS rules, no colors should be present in any class names. Technically, I guess a theme could set `.border-white { border-color: blue }` :scream:.

theskumar replied:
Good catch. I will be updating to more semantic versions. This is reminiscent of old implementation that happens to work ok-ish in the new version.

theskumar replied:
Yes. Setting border-white to blue is possible, technically, and much easier done by setting theme variable "--color-white: blue". 

Instead of doing this, semantic color names should work better for our use case.

bickelj commented on hypha/apply/activity/templates/activity/ui/activity-comment-item.html:
```diff
@@ -1,21 +1,21 @@
             </div>
 
-            <div class="{% if activity.user == request.user %}border-cyan-300{% else %}border-gray-300{% endif %} flex-1 pb-2 bg-white border rounded-sm target:ring-2 target:ring-cyan-500 scroll-m-2"
+            <div class="{% if activity.user == request.user %}border-info/30{% else %}border-base-300{% endif %} flex-1 pb-2 bg-base-100 border rounded-sm target:ring-2 target:ring-cyan-500 scroll-m-2  z-1"
```
Is `ring-cyan-500` obsolete?

theskumar replied:
Generally, the `*-500` colors are works in light and dark mode, so easy to miss, while testing. I think the `-info` color for this makes most sense. But this could also left as it. Ideally, I would say we try avoid all the specific colors.

sandeepsajan0 commented on hypha/apply/projects/templates/application_projects/project_approval_form.html:
```diff
@@ -1,72 +1,78 @@
-                            {% endif %}
+{% block content %}
+    <div class="flex flex-col-reverse gap-4 mt-4 mb-8 lg:flex-row">
+        {% if not approval_form_exists %}
```
I think it should be `{%  if approval_form_exists %}`.
 I checked every project with Project Form is showing a message that Project Form doesn't exist.

theskumar replied:
Thanks. I usually test the UI by put `not` in front. Might have missed revert this one.

theskumar replied:
This should be fixed now.

sandeepsajan0 commented:
1. The project lead is not visible to the finance and contracting role.

<img width="1223" alt="image" src="https://github.com/user-attachments/assets/ef2c532d-0953-4294-b520-acc7277e0eb7" />

2. Contract UI is broken a bit
<img width="1223" alt="image" src="https://github.com/user-attachments/assets/1c891d95-8cc1-46c8-8c7c-c1cd95abcf2e" />

3. Having modals issue with contract upload, it is showing countersigned modal for both options. I guess both options shouldn't be available to a user?
<img width="871" alt="image" src="https://github.com/user-attachments/assets/d0687303-9902-44d3-a2ce-d726b8f26a4e" />
<img width="521" alt="image" src="https://github.com/user-attachments/assets/1dc874f6-fb0a-4264-9b14-08e80243f7ab" />
 
 4. Noticed some nice improvements around invoice, but I think Show/hide toggle button for invoice statuses is misaligned to the bottom instead of the middle. Is it intentional? 
<img width="919" alt="image" src="https://github.com/user-attachments/assets/f5d53f93-9884-49e1-897b-004f34157c92" />
Also statuses info looks a bit compact.
<img width="919" alt="image" src="https://github.com/user-attachments/assets/bac40859-65ea-4d7a-839f-ed3e597c5004" />

theskumar committed:
fix approval_form_exists

theskumar committed:
Fix contracting document row

sandeepsajan0 commented:
1. Submission waiting for review is missing table, while count is showing 1 submission.
<img width="1276" alt="image" src="https://github.com/user-attachments/assets/a75eea45-8192-4209-9707-1e726336952a" />

theskumar committed:
w-4 h-4 -> size-4

theskumar committed:
update project form and it's details

frjo committed:
Hypha theme colours (#4595)

theskumar committed:
tweaks to project detail page

frjo committed:
Fix test for active invoices and minot lint fix.

frjo committed:
Fix test for dashboard.

frjo committed:
Base hyphatheme light on corporate.

theskumar committed:
Remove transition from detail summary

frjo committed:
Fix more tests.

frjo committed:
Fix more tests, part 2.

frjo committed:
Fix more tests, part 3.

frjo committed:
Adjust theme colours.

sandeepsajan0 commented:
Looks like only dark mode is there, it is not converting now to light and dark mode. It might have happened because of latest theme colours commit?

Also, the status bar is not clearly visible.
![image](https://github.com/user-attachments/assets/91aac5e6-4287-4712-9b76-e589495953c4)

sandeepsajan0 committed:
Merge branch 'enhancement/aco-applicant-complete-project' into themes

sandeepsajan0 committed:
Merge remote-tracking branch 'origin/main' into themes

theskumar committed:
use alert component for toast

theskumar committed:
tweak theme and status bar

theskumar committed:
Update the translation cli and fix loader

theskumar committed:
Update invoice detail + fix activity list attribute error

theskumar committed:
invoices -> show rejected

theskumar committed:
Update hero back links

theskumar committed:
theme switcher

theskumar committed:
port over wmd classes

theskumar committed:
convert print.scss to print.css

theskumar committed:
Remove rich-text.css

theskumar committed:
update SOW form page

theskumar committed:
fix SOW detail

frjo commented:
@theskumar A rebase would be good.

wes-otf commented:
It looks like the `Continue editing` button on application preview stopped working and help text for file uploads is a bit weird: 
<img width="582" alt="Screenshot 2025-07-07 at 13 14 03" src="https://github.com/user-attachments/assets/85d4acbe-f92c-45dd-99ad-c79e5f69afef" />

I like the system theme detection that's happening now though! super cool

theskumar committed:
Merge remote-tracking branch 'origin/main' into themes

* origin/main:
  Removed nested htmx triggering submission export on search (#4597)
  Fix Project lead visibility for org memeber like finance (#4599)

theskumar committed:
tweak theme and remove more hard-coded colors

theskumar committed:
reorder theme, fix for theme-switcher

theskumar committed:
Fix screen decision buttons

theskumar committed:
fix broken help-text

theskumar commented:
rebased, and fixed the the help text issues. Also, did some minor tweaks to dark theme. 

From my perspective, the current version is more or less ready to ship, functionally. I'll now focus on more cleanup, while we wrap up the reviews. 

@wes-otf can't reproduce the `Continue Editing` bug, it's simply a link and weird it's not working.

theskumar committed:
remove duplicate variables

theskumar committed:
Remove more custom variables

theskumar committed:
refactor flash-item -> animate-flash

theskumar committed:
Fix typography on application submission landing

frjo commented:
The new theme switcher works and looks great.

I'm unsure about switching place between language and theme switcher. To me language feels more important.

theskumar commented:
> The new theme switcher works and looks great.

glad you liked it. 
 
> I'm unsure about switching place between language and theme switcher. To me language feels more important.

I've used the UX pattern from Django's official documentation website. I think it's debatable, as I've seen language switchers which, in most cases, are either server-side configured or detected automatically by the browser placed at the bottom. Happy to swap back the position. I don't have a very strong preference. 

![Screenshot 2025-07-08 at 1  37 26@2x](https://github.com/user-attachments/assets/120f8a54-1e7f-418d-b8ac-7b18e06ec61b)

theskumar committed:
applicant submission delete button

theskumar committed:
Fix continue editing button

theskumar committed:
Update display of draft application for applicant

theskumar committed:
update how status is displayed in old table view

theskumar committed:
tweak icon size in alert

frjo committed:
Implement updated print styles.
